crates/codegen/syntax_templates/src/rust/cst_visitor.rs

```diff
@@ -1,51 +1,41 @@
 use std::rc::Rc;
 
 use super::cst::*;
-use super::kinds::*;
-use super::language::TextRange;
 
 #[allow(unused_variables)]
 pub trait Visitor<E> {
     fn enter_rule(
         &mut self,
-        kind: RuleKind,
-        range: &TextRange,
-        children: &Vec<Rc<Node>>,
-        node: &Rc<Node>,
-        path: &Vec<Rc<Node>>,
+        node: &Rc<RuleNode>,
+        path: &Vec<Rc<RuleNode>>,
+        source_offset: TextSize,
```

Comment:
I think users are expecting to see ranges, and they will need it to store/pass to editors/linters.
What do you think about providing that instead?

```suggestion
        range: TextRange,
```

Reply:
done